frontend/src/lib/urlQueryPreset.tsx

```diff
@@ -0,0 +1,64 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import queryString from 'query-string'
+
+// This function updates url query string based on page filter selections
+// export function updateBrowserUrlFilterPresets(filters, search, sort) {
+//     // TODO
+// }
+
+/**
+ * transformBrowserUrlToFilterPresets returns preset filters, search text, sorting, current page number and items per page defined in the window.location.search.
+ *
+ * initialFilters - An object of { [key: string]: string[] } entries
+ * Where key matches the filter category id (should be a lowercase string)
+ * Where string[] contains strings of the selected category option values
+ *
+ * initialSearch - A string to use in the table searchbar.
+ *
+ * initialSort - Name of the column to sort on, prefixed with an '-' or '' to denote ascending or descending sorting.
+ *
+ * initialPage - the current page number.
+ *
+ * initialPageSize - How many items to display per page.
+ *
+ * Example usage: ?region=us-east&region=us-south&status=online&status=offline&search=some text goes here&sort=-name&page=5&pageSize=10
+ * Example response: {"initialFilters":{"id":"region","options":["us-east","us-south"]},{"id":"status","options":["online","offline"]},"initialSearch":"some text goes here","initialSort":{"colName":"name","direction":"asc"},"initialPage":5,"initialPageSize":10}
+ * */
+export function transformBrowserUrlToFilterPresets(urlSearch: string) {
+    const initialFilters: { [key: string]: string[] } = {}
+    let initialSearch = ''
+    let initialSort: { colName: string; direction: 'asc' | 'desc' } = { colName: '', direction: 'asc' }
+    let initialPage = 1
+    let initialPageSize = 10
+    const parsedQuery = queryString.parse(urlSearch)
```

Comment:
querystring requires a polyfill from nodejs.
I would be interested in this using URLSearchParams